modules/google-analytics/classes/wp-google-analytics-legacy.php

```diff
@@ -77,6 +77,19 @@ public function insert_code() {
 			return;
 		}
 
+		if ( 'G-' === substr( $tracking_id, 0, 2 ) ) {
+			$this->render_gtag_code( $tracking_id );
+		} else {
+			$this->render_ga_code( $tracking_id );
```

Comment:
If the new code is backwards compatible, how come we would need to keep the old JS code around?

Reply:
The ga side has some custom tracking for 404 and search pages that seemed to be worth preserving. Would gtag back compat cover those?

Reply:
It looks like https://developers.google.com/analytics/devguides/collection/gtagjs/events would cover the 404 tracking, and https://developers.google.com/analytics/devguides/collection/gtagjs/pages#manual_pageviews the search thing?

Reply:
@anomiex I did more reading on this, and in the 404 tracking case, for UA tags, we do this 
```
$custom_vars[] = "_gaq.push(['_trackEvent', '404', document.location.href, document.referrer]);";
```

I'm not entirely sure why we consider this an "event", but based on Google's documentation -- https://developers.google.com/analytics/devguides/collection/gtagjs/events it does not seem to fall under the domain of an "event". Events seem to be user actions.

So I dug some more and am registering it as an "exception" https://developers.google.com/analytics/devguides/collection/gtagjs/exceptions

Lastly, we are doing this for UA tags:

```
		} elseif (
			is_search()
			&& isset( $_REQUEST['s'] )
		) {
			// Set track for searches, if it's a search, and we are supposed to.
			$track['data'] = sanitize_text_field( wp_unslash( $_REQUEST['s'] ) ); // Input var okay.
			$track['code'] = 'search';
		}

		if ( ! empty( $track ) ) {
			$track['url'] = $this->_get_url( $track );
			// adjust the code that we output, account for both types of tracking.
			$track['url']  = esc_js( str_replace( '&', '&amp;', $track['url'] ) );
			$custom_vars[] = "_gaq.push(['_trackPageview','{$track['url']}']);";
		} else {
			$custom_vars[] = "_gaq.push(['_trackPageview']);";
		}
```

Ultimately, what's being done is tracking the URL , but this is all happening by default on every page: https://developers.google.com/analytics/devguides/collection/gtagjs/pages

So I made no changes here.

Reply:
Fine with me on the 404 thing, I only know what I spotted in the docs at a quick glance.

For the search thing, it looks like it's a bit more complicated. The "else" branch of the second `if` there is the default page tracking. But the "then" branch is tracking a URL other than the current one (unless it's just doing a bunch of work to generate the default URL?). But as I said, I'm not very familiar with this stuff so I'll leave the final decision to you and others.